### PR TITLE
add project list command

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-18973]

### CODE REVIEW

File: pkg/cmd/project/project.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/constants/annotations"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdProject(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "project <command>",
+		Aliases: []string{"proj"},
```
Comment (borland) [inline pkg/cmd/project/project.go:16]: we've not put an alias on a toplevel command before. 
@jbristowe what do you think about this? I'm very slightly leaning towards "let's not do that", but not concerned either way.

File: pkg/cmd/project/list/list.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package list
```
Comment (borland) [inline pkg/cmd/project/list/list.go:1]: TODO some tests?

**Review by borland — COMMENTED** [review]
Nice work Ben! I will refrain from approve/reject as I'm supposed to be on nancy :-)

File: pkg/cmd/project/project.go (hunk 1 above)
Comment (jbristowe) [reply pkg/cmd/project/project.go:16]: Agreed. I do not wish to have aliases for top-level commands. Please remove the `env` alias as well.

File: pkg/cmd/project/project.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		Short:   "Manage projects",
+		Long:    `Work with Octopus Deploy projects.`,
+		Example: fmt.Sprintf(heredoc.Doc(`
+			$ %s project list
+			$ %s env ls
```
[the remark this answers is not in the document]
Comment (benPearce1) [reply pkg/cmd/project/project.go:21]: 👍 yes, it was a copy-pasta

File: pkg/cmd/project/project.go (hunk 1 above)
Comment (benPearce1) [reply pkg/cmd/project/project.go:16]: `env` is now removed, no other top-level commands have an alias defined

File: pkg/cmd/project/list/list.go (hunk 2 above)
Comment (benPearce1) [reply pkg/cmd/project/list/list.go:1]: still getting to that

Comment (benPearce1) [reply pkg/cmd/project/list/list.go:1]: actually not required for simple list commands

**Review by domenicsim1 — COMMENTED** [review]
😎 👍

**Review by domenicsim1 — APPROVED** [review]

**Pull request merged**